pkg/apis/v1alpha1/subnetport_types.go

```diff
@@ -0,0 +1,84 @@
+/* Copyright © 2021 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+
+// SubnetPortSpec defines the desired state of SubnetPort
+type SubnetPortSpec struct {
+	// Subnet defines the attachement of the SubnetPort
+	Subnet string `json:"subnet,omitempty"`
+	// SubnetSet defines the attachement of the SubnetPort
+	SubnetSet string `json:"subnetset,omitempty"`
+	// VirtualMachine labels to realize SecurityPolices (VMSelector)
+	VirtualMachine string `json:"VirtualMachine,omitempty"`
+}
+
+// SubnetPortStatus defines the observed state of SubnetPort
+type SubnetPortStatus struct {
+	// Conditions describes current state of SubnetPort
+	Conditions []SubnetPortCondition `json:"conditions,omitempty"`
+	// InterfaceID describes the logical switch port ID owned by the SubnetPort in NSX
+	InterfaceID string `json:"interfaceID,omitempty"`
+	// IPAddresses describes the IP addresses of the SubnetPort
+	IPAddresses []SubnetPortIPAddress `json:"ipAddresses,omitempty"`
+	// MacAddress describes the MAC address of the SubnetPort
+	MacAddress string `json:"macAddress,omitempty"`
+	// ProviderStatus describes a set of NSX logical switches mapping to the subnet/subnetset
+	ProviderStatus SubnetPortProviderStatus `json:"providerStatus,omitempty"`
+}
+
+// SubnetPortCondition defines the condition of policy.
+type SubnetPortCondition struct {
+	// Type defines the type of condition.
+	Type SubnetPortStatusCondition `json:"type,omitempty"`
+	// Status shows the status of condition, one of True or False.
+	Status corev1.ConditionStatus `json:"status,omitempty"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about the condition.
+	Message string `json:"message,omitempty"`
+}
+
+// SubnetPortStatusCondition is an aspect of operator state.
+type SubnetPortStatusCondition string
+
+type SubnetPortIPAddress struct {
+	Gateway string `json:"gateway,omitempty"`
+	IP string `json:"ip,omitempty"`
+	SubnetMask string `json:"subnetMask,omitempty"`
```

Comment:
Could we add the below validation for subnetMask?
// SubnetMask defines an IPv4 range in the form <ip>/<bits>.
// +kubebuilder:validation:Pattern="^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\\/((3[0-2])|([1-2][0-9])|[1-9])$"